lib/train/transports/local.rb

```diff
@@ -17,17 +17,48 @@ def connection(_ = nil)
     end
 
     class Connection < BaseConnection
+      require 'json'
+      require 'base64'
+      require 'securerandom'
+
       def initialize(options)
         super(options)
         @cmd_wrapper = nil
         @cmd_wrapper = CommandWrapper.load(self, options)
+        @pipe = acquire_named_pipe if @platform.windows?
+      end
+
+      def acquire_named_pipe
+        pipe_name = "inspec_#{SecureRandom.hex}"
+        pipe_location = "//localhost/pipe/#{pipe_name}"
+        start_named_pipe_server(pipe_name) unless File.exist?(pipe_location)
+
+        # Try to acquire pipe for 10 seconds with 0.1 second intervals.
+        # This allows time for PowerShell to start the pipe
+        pipe = nil
+        100.times do
+          begin
+            pipe = open(pipe_location, 'r+')
+            break
+          rescue
+            sleep 0.1
+          end
+        end
+        fail "Could not open named pipe #{pipe_location}" if pipe.nil?
+
+        pipe
       end
 
       def run_command(cmd)
-        cmd = @cmd_wrapper.run(cmd) unless @cmd_wrapper.nil?
-        res = Mixlib::ShellOut.new(cmd)
-        res.run_command
-        CommandResult.new(res.stdout, res.stderr, res.exitstatus)
+        if defined?(@pipe)
+          res = run_powershell_using_named_pipe(cmd)
+          CommandResult.new(res['stdout'], res['stderr'], res['exitstatus'])
+        else
```

Comment:
I think this is problematic, we check if a pipe is available. Lets assume we are on a windows platform and pipe creation failed, then we need to implement a fallback. Essentially we need to decide:
```
if  os.windows?
  if pipe
    # use pipe 
  else
    # fallback, which may be slower but that is okay
  end
else 
  # linux / mac / unix
end 
```

Reply:
I'll bring back the old functionality and then wrap `acquire_named_pipe` in a `begin/rescue`. That way if the pipe fails for whatever reason, it will default to the way things run now.